### DISCUSSION TITLE
Channels stop communicating without error

### DISCUSSION THREAD

**Discussion by mmembrino-glytec** [nextgenhealthcare/connect#5419] [Q&A]
Hi - We are running MIRTH Connect 3.10.1 in Azure.   We have many different customers that send HL7 feeds into our MIRTH engine.   We use a point-to-point VPN per customer/hospital -- The MIRTH DB is MSSQL.    The channels takes the HL7 data and formats the data appropriately for use by SQL stored procedures to update our product DBs which are also SQL.

For the past 2 weeks we have been having channels that are processing just fine - and then just hang/drop.    The channel remains in a starting/connected status -- but transactions just continue to queue.    All interfaces affected are ADT interfaces.   We also have many LIS, results, MAR, etc - and none of these are affected).    Attempting to stop the channel cleanly are not successful - we eventually have to halt the channel.     Getting the channel back up and functioning varies --  Some have been able to restart with a simple restart.     Others we've had to redeploy the channel.     Redeploying puts the channel into an "undeploying" state - which does not successfully undeploy - which also requires a halt.     After the halt is completed, the deploy of the channel resumes and completes successfully.     Upon review of the message logs -- there is nothing in the log to indicate error.     The transaction/message that appeared to be "hung" - processed cleanly upon restart.     We reviewed the console/server logs - Mirth.log*  and there is nothing in there with the exception of errors thrown when we halt the interface.    Nothing prior to the halt.

We've asked our network team to review and they have identified nothing in the network logs.   No high CPU,  no low disk space, not close to being out of RAM.   
We've asked out database team to review and they haven't identified anything in the SQL logs.   No errors, no blocks, and no deadlocks.

We've had 5 different customer ADTs affected last week - Monday, Tuesday, Wednesday.   We asked the support team to not attempt to restart the channels until we had a chance to review.    We did not have another occurrence until today.     We did real time review and came up with nothing.

Are there other logs to check other than the logs here:  opt/mirthconnect/logs/mirth.log*
What else can we troubleshoot?    This seems to be one of the dreaded issues where something is obviously happening because the channel dies - but we're not finding anything in the logs to assist with the issue.    The only commonality is that it is ADT -- but not all customers are having the problem.

Any help would be appreciated.
Thanks!
Maria

We activated extended events for more logging and we did not see

**Comment by pacmano1** [comment]
I think your troubleshooting steps seem on target.   

What is the downstream thing behind the queued messages?   Just taking ADT and sending to SQL server?

**Reply by mmembrino-glytec** [reply]
Yes - We just take the ADT data and format the data so that it can be used in the SQL stored procedures.   There is some business processing in there that says stuff like - If event type = A13 do blah,   If eventype = A40 - do blah.    But all of the business processing makes sure the data is prepped for processing by the SQL stored procedures.    We have asked the DBAs to rule out SQL issues - and they are investigating that while we continue to look for causes in MIRTH.

Are there any other logs for us to research that you are aware of -  other than the Mirth Connect logs?

**Reply by pacmano1** [reply]
Sort of guessing here,  jdbc pool problem perhaps?

**Reply by jonbartels** [reply]
> Are there any other logs for us to research that you are aware of - other than the Mirth Connect logs?

I would do a few things:
1. Enable debug level logging in MC. It'll be VERY noisy but add it
2. Enable monitoring on SQL server for connection counts from the MC DB user, are connections dropping or being closed?
3. Enable monitoring in MC for basic health checks - ping ports, HTTP GET to the JNLP URL, etc
4. Enable monitoring in MC for message volume - There are commercial tools but writing something that hits the dashboard API to watch for sent/received counts not changing for some period and alerting will help catch it
5. Enable monitoring in MC for JMX metrics (heap and thread count) - Most tools like Datadog or Prometheus or Grafana can handle this
6. Have a clear written plan to recover the instance while you figure out whats going on. Not crashing is best but can you restart fast and recover

How are you running MC? Is it a service in Linux or Windows? AWS/Azure or a regular VM?
